storage/diff_scheme/diff_manager.cpp

```diff
@@ -0,0 +1,125 @@
+#include "storage/diff_scheme/diff_manager.hpp"
+#include "storage/diff_scheme/diff_scheme_checker.hpp"
+
+#include "generator/mwm_diff/diff.hpp"
+
+#include "platform/platform.hpp"
+
+#include "coding/internal/file_data.hpp"
+
+#include "base/assert.hpp"
+
+namespace storage
+{
+namespace diffs
+{
+void Manager::Load(LocalMapsInfo && info)
+{
+  {
+    std::lock_guard<std::mutex> lock(m_mutex);
+    m_localMapsInfo = std::move(info);
+  }
+
+  m_workerThread.Push([this]
+  {
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    NameFileInfoMap const diffs = Checker::Check(m_localMapsInfo);
```

Comment:
Do we really need to call Check with locked mutex? I suggest you copy m_localMapsInfo to temprorary variable and call Check() then.